master/internal/task/allocation.go

```diff
@@ -813,48 +818,53 @@ func (a *Allocation) kill(ctx *actor.Context, reason string) {
 }
 
 func (a *Allocation) registerProxies(ctx *actor.Context, addresses []cproto.Address) {
-	cfg := a.req.ProxyPort
-	if cfg == nil {
-		return
-	}
-	if len(a.resources) > 1 {
-		// We don't support proxying multi-reservation allocations.
-		ctx.Log().Warnf("proxy for multi-reservation allocation aborted")
+	// For multi-reservation allocations, proxies are only setup for rank=0 (i.e. the chief).
+	if len(a.req.ProxyPorts) == 0 {
 		return
 	}
 
-	for _, address := range addresses {
-		// Only proxy the port we expect to proxy. If a dockerfile uses an EXPOSE command,
-		// additional addresses will appear her, but currently we only proxy one uuid to one
-		// port, so it doesn't make sense to send multiple proxy.Register messages for a
-		// single ServiceID (only the last one would work).
-		if address.ContainerPort != cfg.Port {
-			continue
-		}
+	if len(a.req.ProxyPorts) > 0 {
+		for _, address := range addresses {
+			// Only proxy the port we expect to proxy. If a dockerfile uses an EXPOSE command,
+			// additional addresses will appear her, but currently we only proxy one uuid to one
+			// port, so it doesn't make sense to send multiple proxy.Register messages for a
+			// single ServiceID (only the last one would work).
+			var pcfg *sproto.ProxyPortConfig
+			for _, cfg := range a.req.ProxyPorts {
+				if address.ContainerPort == cfg.Port {
+					pcfg = cfg
+				}
+			}
+			if pcfg == nil {
+				continue
+			}
 
-		// We are keying on allocation id instead of container id. Revisit this when we need to
-		// proxy multi-container tasks or when containers are created prior to being
-		// assigned to an agent.
-		ctx.Ask(ctx.Self().System().Get(actor.Addr("proxy")), proxy.Register{
-			ServiceID: cfg.ServiceID,
-			URL: &url.URL{
-				Scheme: "http",
-				Host:   fmt.Sprintf("%s:%d", address.HostIP, address.HostPort),
-			},
-			ProxyTCP:        cfg.ProxyTCP,
-			Unauthenticated: cfg.Unauthenticated,
-		})
-		a.proxies = append(a.proxies, cfg.ServiceID)
+			// We are keying on allocation id instead of container id. Revisit this when we need to
+			// proxy multi-container tasks or when containers are created prior to being
+			// assigned to an agent.
+			ctx.Ask(ctx.Self().System().Get(actor.Addr("proxy")), proxy.Register{
+				ServiceID: pcfg.ServiceID,
+				URL: &url.URL{
+					Scheme: "http",
+					Host:   fmt.Sprintf("%s:%d", address.HostIP, address.HostPort),
+				},
+				ProxyTCP:        pcfg.ProxyTCP,
+				Unauthenticated: pcfg.Unauthenticated,
+			})
+			ctx.Log().Debugf("registered proxy id: %s, tcp: %v\n", pcfg.ServiceID, pcfg.ProxyTCP)
+			a.proxies = append(a.proxies, pcfg.ServiceID)
+		}
 	}
 
-	if len(a.proxies) != 1 {
-		ctx.Log().Errorf("did not proxy as expected %v (found addrs %v)", len(a.proxies), addresses)
+	if len(a.proxies) != len(a.req.ProxyPorts) {
+		ctx.Log().Errorf(
```

Comment:
can you send this to the trial log? seems important, now, that an end user sees it.